### PR TITLE
Upgrade Spark to 2.0.2

### LINKED ISSUE #262: Spark 2.0 Pyspark Image
For latest versions of Apache Toree (incubating) we need a docker-stacks build that has Spark 2.0 in it.
Development is going on here https://github.com/apache/incubator-toree/pull/56

```bash
RUN export APACHE_SPARK_VERSION=2.0.0 \
   wget http://apache.claz.org/spark/spark-$(APACHE_SPARK_VERSION)/spark-$(APACHE_SPARK_VERSION)-bin-hadoop2.6.tgz && \
   tar xzf spark-$(APACHE_SPARK_VERSION)-bin-hadoop2.6.tgz -C /usr/local && \
   cd /usr/local && \
   ln -s spark-$(APACHE_SPARK_VERSION)-bin-hadoop2.6 spark
```

Due to Toree using a more recent version of Akka we also require JDK8.

Pinging @parente @lbustelo

### LINKED ISSUE #288: spark 2.0 jupyter/all-spark-notebook image anyone?
I'm continuing to have problems connecting to S3 on remote spark clusters (see issue #277) but spark 2.0 seems to solve the problem on the backend.  Now I need a jupyter toree built against spark 2.0 but that's proving to be more difficult to build than I would have hoped.

does anyone have a spark 2.0 jupyter/all-spark-notebook docker image to share?
thanks in advance!

### PR DESCRIPTION
**Pull request by parente**
Closes #288 #262 #301.

Might help with #277 and #272 as well.

### CODE REVIEW

**Comment by aliabbasjp** [comment]
so can I use this dockerfile and build my own local SPARK 2 docker ? @parente

**Comment by parente** [comment]
@aliabbasjp yes but the version of Apache toree that is currently released will not work in the image.

**Comment by Plippe** [comment]
Just a ping as it seems Toree is now compatible with Spark 2.0 (based on the merge status of https://github.com/apache/incubator-toree/pull/56, and the official [readme](https://github.com/apache/incubator-toree/blob/master/README.md#version)).

**Comment by parente** [comment]
Master is compatible, but we're waiting on a release: https://github.com/jupyter/docker-stacks/issues/288#issuecomment-249580477

**Comment by parente** [comment]
Going with the dev release of the master branch which should be 2.0.x compatible:

https://dist.apache.org/repos/dist/dev/incubator/toree/0.2.0/snapshots/dev1/toree-pip/

**Comment by parente** [comment]
Build works. I did some manual QA here.

Should be GTG after a review of the changes.

File: all-spark-notebook/README.md [hunk 1]
```diff
@@ -223,7 +219,7 @@ You may customize the execution of the Docker container and the Notebook server
...
 * `-v /some/host/folder/for/server.pem:/home/jovyan/.local/share/jupyter/notebook.pem` - Mounts a SSL certificate plus key for `USE_HTTPS`. Useful if you have a real certificate for the domain under which you are running the Notebook server.
-* `-p 4040:4040` - Opens the port for the [Spark Monitoring and Instrumentation UI](http://spark.apache.org/docs/latest/monitoring.html). Note every new spark context that is created is put onto an incrementing port (ie. 4040, 4041, 4042, etc.), and it might be necessary to open multiple ports. `docker run -d -p 8888:8888 -p 4040:4040 -p 4041:4041 jupyter/all-spark-notebook`
+* `-p 4040:4040` - Opens the port for the [Spark Monitoring and Instrumentation UI](http://spark.apache.org/docs/latest/monitoring.html). Note every new Spark context that is created is put onto an incrementing port (ie. 4040, 4041, 4042, etc.), and it might be necessary to open multiple ports. `docker run -d -p 8888:8888 -p 4040:4040 -p 4041:4041 jupyter/all-spark-notebook`
```
Comment (mariusvniekerk) [inline all-spark-notebook/README.md]: You can specify the port that the spark ui runs on by setting the `spark.ui.port` configuration option.

Comment (parente) [inline all-spark-notebook/README.md]: Added a note.

**Comment by parente** [comment]
Rebased.

**Comment by parente** [comment]
In tag `2410ad57203a`.

**Pull request merged**